src/pages/index.js

```diff
@@ -1,122 +1,187 @@
+);
 editFormValidator.enableValidation();
-const addFormValidator = new FormValidator(config, addCardFormElement);
-addFormValidator.enableValidation();
 
+const addFormValidator = new FormValidator(
+  constants.config,
+  constants.addCardFormElement
+);
+addFormValidator.enableValidation();
+addFormValidator.toggleButtonState();
+const profileAvatarFormValidator = new FormValidator(
+  constants.config,
+  constants.profileAvatarModalForm
+);
+profileAvatarFormValidator.enableValidation();
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+const cardPopup = new PopupWithForm(
+  "#add-card-modal",
+  // handleAddCardFormSubmit
+  (link) => {
+    cardPopup.renderLoading(true);
+    api
+      .addCard(link)
+
+      .then((res) => {
+        cardSection.addItem(res);
+        cardPopup.close();
+        cardPopup.renderLoading(false);
+      })
+      .then(() => {})
+
+      .catch((err) => {
+        console.error(`Error ${err}`);
+      });
+  }
+);
 cardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const userInfoPopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileFormSubmit
-);
-const cardSection = new Section(initialCards, renderCard, cardListEl);
-cardSection.renderItems();
+const profilePopup = new PopupWithForm("#profile-avatar-modal", (url) => {
+  // handles the avatar picture submit
+  profilePopup.renderLoading(true);
+  api
+    .setUserAvatar(url)
+    .then((res) => {
+      profileAvatarFormValidator.toggleButtonState();
+      userInfo.setUserAvatar(res);
+      profilePopup.close();
+      profilePopup.renderLoading(false);
+    })
+    .catch((err) => {
+      console.error(`Error ${err}`);
+    });
+});
+profilePopup.setEventListeners();
 
+const userInfoPopup = new PopupWithForm("#profile-edit-modal", (values) => {
+  userInfo.setUserInfo(values);
+  userInfoPopup.renderLoading(true);
+  api
+    .setUserInfo(values)
+    .then(() => {
+      editFormValidator.toggleButtonState();
+      userInfoPopup.close();
+      userInfoPopup.renderLoading(false);
+    })
+    .catch((err) => {
+      console.error(`Error ${err}`);
+    });
+});
 userInfoPopup.setEventListeners();
-//----------------functions
 
 function renderCard(cardData) {
-  const card = new Card(cardData, cardTemplate, handleImageClick);
-  return card.getView();
-}
+  const card = new Card(
+    cardData,
+    constants.cardTemplate,
+
+    //handles the image click
+    function imageBtnClikc(name, link) {
+      imagePopup.open(name, link);
+    },
+    //handles the delete button click
+    function deleteBtnClick(cardInstance) {
+      cardDeleteModal.open();
+
+      cardDeleteModal.setSubmitAction(() => {
+        api.deleteCard(cardInstance.getId()).then(() => {
+          cardDeleteModal.close();
+        });
+        cardInstance.removeCard();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.


```js
        api.deleteCard(cardInstance.getId()).then(() => {
          cardDeleteModal.close();
          cardInstance.removeCard();
        }).catch(...)  // don't forget about `catch`

```